src/sortStudents.ts

```diff
@@ -1,16 +1,56 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string
+  age: number
+  married: boolean
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const averageGrade = (grades: number[]): number => {
+    return grades
+      .reduce((sum: number, value: number) => sum + value, 0) / grades.length;
+  };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const copy = [...students];
+
+  switch (sortBy) {
+    case 'grades': copy.sort((a: Student, b: Student) => {
+      return order === 'asc'
+        ? averageGrade(a.grades) - averageGrade(b.grades)
+        : averageGrade(b.grades) - averageGrade(a.grades);
+    }); break;
```

Comment:
if you have return you don't need to use break
```suggestion
    });
```